sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDataWritingCommandExec.scala

```diff
@@ -84,10 +84,8 @@ object GpuDataWritingCommand {
       if (fs.exists(filePath) &&
           fs.getFileStatus(filePath).isDirectory &&
           fs.listStatus(filePath).length != 0) {
-        TrampolineUtil.throwAnalysisException(
-          s"CREATE-TABLE-AS-SELECT cannot create table with location to a non-empty directory " +
-              s"${tablePath} . To allow overwriting the existing non-empty directory, " +
-              s"set '$allowNonEmptyLocationInCTASKey' to true.")
+        AnalysisExceptionShim.throwException("_LEGACY_ERROR_TEMP_1241",
```

Comment:
We should not be throwing literal magic strings.  This is really fragile and likely to have a typo that won't be caught by the compiler and thus likely to not do what we want at runtime.  The Spark code is not doing this in all the various places in Catalyst, instead those are calling QueryExecutionErrors with a descriptive method name, and _that_ is doing the specifics of throwing a specific message.

That's what I mentioned before:
> We have our own wrappers to that in shims, so maybe the best way to fix this is to migrate these use cases to those already shim'd error interfaces

We shouldn't be creating a new shim class here, but rather using the one we already have, i.e.: RapidsErrorUtils.  For example, for this case here we should add a `createTableAsSelectWithNonEmptyDirectoryError` method to the RapidsErrorUtils shim class.  On modern Spark versions, it would simply delegate to Spark's QueryCompilationErrors.  On older Spark versions, the RapidsErrorUtils shim would throw an AnalysisException with an appropriate message string it constructs based on the arguments.

In short, see what the new Spark code is doing in these areas.  I suspect none of them are throwing AnalysisException but rather calling an error utility class like QueryExecutionErrors or QueryCompilationErrors.  We should call the same methods on Spark versions that have those methods available, not throw a derived class.

Reply:
Ah, I was fixated on solving it using the errorClass even for older versions of Spark.

Reply:
Yeah the shim logic should almost always be "do what this version of Spark does for this" whether that's using an error class abstraction or throwing a raw AnalysisException with a string message.

Reply:
OK. I think I got it right this time. This will be painful to review PTAL